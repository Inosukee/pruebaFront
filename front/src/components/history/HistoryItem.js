import React from 'react'

export const HistoryItem = ({ name, last = false }) => {
  return (

    !last ?
      (
        <li>
          <a href="!#">{name}</a>
          <div className='history__svg'>
            <svg xmlns="http://www.w3.org/2000/svg" width="6" height="8"><path fill="none" stroke="#666" d="M1 0l4 4-4 4"></path></svg>
          </div>
        </li>

      )
      :
      (
        <li>
          <a href="!#">{name}</a>
        </li>
      )
  )
}
