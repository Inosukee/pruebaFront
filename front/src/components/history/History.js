import React from 'react'
import { HistoryItem } from './HistoryItem'

export const History = ({ categories }) => {
    return (
        <div className='history__content'>
            <ul className='history__history'>
                {
                    categories &&
                    categories.map((history, i, { length }) => {
                        if (i < length - 1) {
                            return (<HistoryItem key={history.id}
                                {...history} />)
                        } else {
                            return (<HistoryItem last={true} key={history.id}
                                {...history} />)
                        }
                    })
                }
            </ul>
        </div>
    )
}
