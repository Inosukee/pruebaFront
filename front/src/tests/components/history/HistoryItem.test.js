import React from 'react'
import { shallow } from 'enzyme';
import { HistoryItem } from '../../../components/history/HistoryItem';


const nameHistory = 'Test'

describe('Pruebas en <HistoryItem />', () => {
  test('debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<HistoryItem name={nameHistory} />)
    expect(wrapper.find('.history__svg').exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})