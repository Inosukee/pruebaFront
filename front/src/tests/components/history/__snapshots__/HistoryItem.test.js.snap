// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HistoryItem /> debe mostrar el componente correctamente 1`] = `
<li>
  <a
    href="!#"
  >
    Test
  </a>
  <div
    className="history__svg"
  >
    <svg
      height="8"
      width="6"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 0l4 4-4 4"
        fill="none"
        stroke="#666"
      />
    </svg>
  </div>
</li>
`;
